client/js/_author/components/assessments/question_types/multiple_choice.jsx

```diff
@@ -1,99 +1,85 @@
 import React    from 'react';
-
-export default function (props) {
-
-
-  return (
-    <div className="c-question__answers c-question__answers--maintain">
-
-      <div className="c-answer is-active">
-        <div className="c-input">
-          <div className="c-radio">
-            <input type="radio" id="radio" name="radio" tabIndex="0" />
-            <label htmlFor="radio"></label>
-          </div>
-
-          <label htmlFor="option1"></label>
-          <div className="c-input__contain">
-            <input className="c-text-input c-text-input--small c-wysiwyg" id="option1" type="text" placeholder="Option 1" tabIndex="0" />
-            <div className="c-input__bottom"></div>
-          </div>
-
-          <div className="c-answer__icons">
-            <button className="c-answer__icons__spacer" tabIndex="0">
-              <i className="material-icons">arrow_upward</i>
-            </button>
-            <button className="c-answer__icons__spacer" tabIndex="0">
-              <i className="material-icons">arrow_downward</i>
-            </button>
-            <button className="c-answer__icons__spacer" tabIndex="0">
-              <i className="material-icons">close</i>
-            </button>
-          </div>
-        </div>
-
-        <div className="c-input c-input-label--left c-feedback">
-          <label htmlFor="feedback1">Feedback</label>
-          <div className="c-input__contain">
-            <input className="c-text-input c-text-input--smaller c-wysiwyg" id="feedback1" type="text" tabIndex="0" />
-            <div className="c-input__bottom"></div>
-          </div>
-        </div>
+import _        from 'lodash';
+import Option   from './multiple_choice_option';
+import Add      from './add_option';
+
+export default class multipleChoice extends React.Component {
+  static propTypes = {
+    item: React.PropTypes.shape({
+      answers: React.PropTypes.arrayOf(React.PropTypes.shape),
+      id: React.PropTypes.string,
+      question: React.PropTypes.shape({
+        choices: React.PropTypes.arrayOf(React.PropTypes.shape({})),
+      }),
+    }).isRequired,
+    updateItem: React.PropTypes.func.isRequired,
+    updateChoice: React.PropTypes.func.isRequired,
+  };
+
+  hasQuestions() {
+    const { question } = this.props.item;
+    return question
+    && question.choices
+    && question.choices.length;
+  }
+
+  markedForDeletion(choice) {
+    const { question } = this.props.item;
+    const deleteChoice = _.find(question.choices, { id: choice.id });
+    deleteChoice.delete = true;
+    return question.choices;
+  }
+
+  deleteChoice(choice) {
+    this.props.updateItem({
+      question: {
+        choices: this.markedForDeletion(choice)
+      }
+    });
+  }
+
+  moveChoice(choice, up) {
```

Comment:
Maybe have the options know their index, so you don't have to do the find stuff?